### PR TITLE
feat: allow api role to be assumed by anyone in LINZ organization

### PR DESCRIPTION
**Pull request by MitchellPaff**
## Reference

[Code review checklist](CODING.md#Code-review-checklist)

### CODE REVIEW

File: infrastructure/constructs/api.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def __init__(  # pylint: disable=too-many-arguments,too-many-locals
             self,
             "api-users-role",
             role_name=Resource.API_USERS_ROLE_NAME.resource_name,
-            assumed_by=principal,
+            assumed_by=aws_iam.OrganizationPrincipal("o-g9kpx6ff4u"),
```
Comment (l0b0) [inline infrastructure/constructs/api.py:40]: Do we need to also replace the other use of `principal`, in the S3 users role? (At which point we can get rid of the `principal` parameter.)

Comment (billgeo) [reply infrastructure/constructs/api.py:40]: Just to clarify.

s3-users role should be able to be assumed by anyone at LINZ. So if you can achieve that through this same mechanism that's all good.

**Review by billgeo — COMMENTED** [review]
Can you also look at the USAGE.md for any changes related to how people will assume roles?

File: USAGE.md [hunk 2]
```diff
@@ -117,6 +117,9 @@ Example of AWS service account authentication and authorization in to Geostore u
   export AWS_PROFILE=geostore-users
   ```
 
+Lambda and S3 access is also enabled by default for any AWS accounts in the LINZ organisation. This
+is by assuming the 'api-users' or 's3_users' role from the external AWS account.
```
Comment (l0b0) [inline USAGE.md:121]: Should `s3_users` be `s3-users`?

**Review by l0b0 — DISMISSED** [review]

**Review by l0b0 — APPROVED** [review]

**Pull request merged**